server/src/internalClusterTest/java/org/elasticsearch/snapshots/DataStreamsSnapshotsIT.java

```diff
@@ -0,0 +1,224 @@
+
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+public class DataStreamsSnapshotsIT extends AbstractSnapshotIntegTestCase {
+
+    public void testSnapshotAndRestore() throws Exception {
+        Client client = client();
+
+        Path location = randomRepoPath();
+        createRepository("repo", "fs", location);
+
+        DataStreamIT.createIndexTemplate("t1", "@timestamp", "ds", "other-ds");
+
+        CreateDataStreamAction.Request request = new CreateDataStreamAction.Request("ds");
+        AcknowledgedResponse response = client.admin().indices().createDataStream(request).get();
+        assertTrue(response.isAcknowledged());
+
+        request = new CreateDataStreamAction.Request("other-ds");
+        response = client.admin().indices().createDataStream(request).get();
+        assertTrue(response.isAcknowledged());
+
+        Map<String, Integer> source = Collections.singletonMap("@timestamp", 123);
+        IndexResponse indexResponse = client.prepareIndex("ds")
+            .setOpType(DocWriteRequest.OpType.CREATE)
+            .setSource(source)
+            .get();
+        assertEquals(DocWriteResponse.Result.CREATED, indexResponse.getResult());
+
+        CreateSnapshotResponse createSnapshotResponse = client.admin().cluster()
+            .prepareCreateSnapshot("repo", "snap")
+            .setWaitForCompletion(true)
+            .setIndices("ds")
+            .setIncludeGlobalState(false)
+            .get();
+
+        RestStatus status = createSnapshotResponse.getSnapshotInfo().status();
+        assertEquals(RestStatus.OK, status);
+
+        GetSnapshotsResponse snapshot = client.admin().cluster().prepareGetSnapshots("repo").setSnapshots("snap").get();
+        List<SnapshotInfo> snap = snapshot.getSnapshots("repo");
+        assertEquals(1, snap.size());
+        assertEquals(Collections.singletonList(".ds-ds-000001"), snap.get(0).indices());
+
+        assertTrue(client.admin().indices().deleteDataStream(new DeleteDataStreamAction.Request("ds")).get().isAcknowledged());
+
+        RestoreSnapshotResponse restoreSnapshotResponse = client.admin().cluster()
+            .prepareRestoreSnapshot("repo", "snap")
+            .setWaitForCompletion(true)
+            .setIndices("ds")
+            .get();
+
+        assertEquals(1, restoreSnapshotResponse.getRestoreInfo().successfulShards());
+
+        GetResponse getResponse = client.prepareGet(".ds-ds-000001", indexResponse.getId()).get();
+        assertEquals(source, getResponse.getSourceAsMap());
+
+        GetDataStreamAction.Response ds = client.admin().indices().getDataStreams(new GetDataStreamAction.Request("ds")).get();
+        assertEquals(1, ds.getDataStreams().size());
+        assertEquals(1, ds.getDataStreams().get(0).getIndices().size());
+        assertEquals(".ds-ds-000001", ds.getDataStreams().get(0).getIndices().get(0).getName());
```

Comment:
same here and a couple other places below

Reply:
Done